From f8859558b515c122027171a92acda18435059970 Mon Sep 17 00:00:00 2001
From: root <root@allinone.cloud.local>
Date: Mon, 21 Nov 2016 04:59:54 +0000
Subject: [PATCH] avoid unnecessary OVS interface list retries

It seems that the change introduced by cd78c7c triggers unnecessary calls
to ovs-vsctl. Not only this pollute the logs with a lot of exceptions,
but it also slows down the ovs configuration done by neutron-openvswitch.

This patch should keep the initial intention of retrying on unassigned
ofport instead of treating it as a failure (Open vSwitch will return '[]'
when querying an interface's ofport when the ofport has not yet been
assigned) and avoid retrying in the case Open vSwitch command exits with
an error.
---
 usr/lib/python2.7/dist-packages/neutron/agent/common/ovs_lib.py | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/usr/lib/python2.7/dist-packages/neutron/agent/common/ovs_lib.py b/usr/lib/python2.7/dist-packages/neutron/agent/common/ovs_lib.py
index e5034d2..05d76a6 100644
--- a/usr/lib/python2.7/dist-packages/neutron/agent/common/ovs_lib.py
+++ b/usr/lib/python2.7/dist-packages/neutron/agent/common/ovs_lib.py
@@ -253,7 +253,7 @@ class OVSBridge(BaseOVS):

     @_ofport_retry
     def _get_port_ofport(self, port_name):
-        return self.db_get_val("Interface", port_name, "ofport")
+        return self.db_get_val("Interface", port_name, "ofport", check_error=True)

     def get_port_ofport(self, port_name):
         """Get the port's assigned ofport, retrying if not yet assigned."""
@@ -264,6 +264,8 @@ class OVSBridge(BaseOVS):
             LOG.exception(_LE("Timed out retrieving ofport on port %(pname)s. "
                               "Exception: %(exception)s"),
                           {'pname': port_name, 'exception': e})
+        except Exception:
+            LOG.error("Failed to retrieve ofport on port %s.", port_name)
         return ofport

     def get_datapath_id(self):
--
1.9.1
